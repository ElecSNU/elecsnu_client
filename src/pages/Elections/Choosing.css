#choosing {
    height: 100%;
    padding: 5%;
    width: 100%;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: center;
}

#choosing > * {
    width: 100%;
}

.candidate {
    padding: 2.5%;
    margin: 2.5%;
    transition: all 0.7s;
    border: 5px solid transparent;
    cursor: pointer;
}

.candidate h1 {
    transition: all 0.7s;
}

.candidate:hover,
.candidate.chosen {
    border: 5px solid var(--accent1);
}

.candidate:hover h1,
.candidate.chosen h1 {
    color: var(--accent1);
}

#choosing .submit-button {
    width: 50%;
}
