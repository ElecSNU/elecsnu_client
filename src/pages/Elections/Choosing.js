import React, { useState } from 'react';
import useFirestore from '../../hooks/useFirestore';
import { fire_store } from '../../configs/firebase/config';
import { toast } from 'react-toastify';
import useLoader from '../../hooks/useLoader';

import Share from './Share';

import './Choosing.css';

const Choosing = ({ candidates }) => {
    const [chosen, setChosen] = useState(0);
    const electionDocs = useFirestore('newelections').docs;
    const showLoader = useLoader();
    const [done, setDone] = useState(false);
    const [electionName, setElectionName] = useState('');

    const castVote = () => {
        showLoader(true);
        const electionId = window.location.pathname.split(
            '/'
        )[3];
        let electionDetails = {};
        electionDocs.forEach((e) => {
            if (e.id === electionId) {
                electionDetails = e;
                return;
            }
        });
        setElectionName(electionDetails.election_name);

        let userGender = window.localStorage.getItem(
            'voter_gender'
        );
        let userDept = window.localStorage.getItem(
            'voter_dept'
        );
        let userBatch = window.localStorage.getItem(
            'voter_batch'
        );
        let userRoll = window.localStorage.getItem(
            'user_roll'
        );

        electionDetails.voters = [
            ...electionDetails.voters,
            Number(userRoll),
        ];
        electionDetails.election_votes = [
            ...electionDetails.election_votes,
            {
                voter_gender: userGender,
                groups: `${userDept},Batch${userBatch}`,
                candidate_chosen: chosen,
            },
        ];

        fire_store
            .collection('newelections')
            .doc(electionId)
            .set(electionDetails);

        showLoader(false);

        toast.success('Successfully casted vote!');

        setDone(true);
    };

    return (
        <>
            {done ? (
                <Share electionName={electionName} />
            ) : (
                <></>
            )}
            <div id='choosing'>
                <div className='heading-text capital-text center-text foreground-back'>
                    Who would you want to vote for?
                </div>
                <div className='candidates'>
                    {candidates.map((candidate, index) => {
                        return (
                            <div
                                className={`candidate background-1 border-radius-15 ${
                                    index === chosen
                                        ? 'chosen'
                                        : ''
                                }`}
                                key={index}
                                onClick={() =>
                                    setChosen(index)
                                }
                            >
                                <h1 className='capital-text subheading-text foreground-light center-text'>
                                    {candidate}
                                </h1>
                            </div>
                        );
                    })}
                </div>
                <button
                    className='submit-button'
                    onClick={castVote}
                >
                    CAST VOTE
                </button>
            </div>
        </>
    );
};

export default Choosing;
