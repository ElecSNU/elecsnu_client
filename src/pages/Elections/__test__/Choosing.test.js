import { render, screen, fireEvent, cleanup } from '@testing-library/react';
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Choosing from './../Choosing'
import { toast } from 'react-toastify';
import React, { useEffect, useState } from 'react';
// import Elections from './../Elections';
// import ParticleBackground from '../../components/ParticleBackground/ParticleBackground';


test('Tests the rendering of the Choosing page', () => {
    const div = document.createElement("div")
    const tree = renderer.create(<index/>).toJSON();
    expect(tree).toMatchSnapshot(); 
   });