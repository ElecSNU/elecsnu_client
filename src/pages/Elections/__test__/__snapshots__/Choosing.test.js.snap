// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the rendering of the Choosing page 1`] = `<index />`;
